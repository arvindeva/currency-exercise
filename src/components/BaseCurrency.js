import React from 'react';
import styled from 'styled-components';

const StyledBaseCurrency = styled.div.attrs({
  className: 'ui segment'
})`
  .ui .input input {
    text-align: right;
    font-size: 36px;
    font-weight: 600;
  }

  input[type='number']::-webkit-inner-spin-button,
  input[type='number']::-webkit-outer-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  p {
    font-style: italic;
  }

  span {
    color: red;
    cursor: pointer;
    text-align: center;
  }

  .header {
    display: flex;
    justify-content: space-between;
    align-items: center;

    h1 {
      margin-top: 0px;
      margin-bottom: 0px;
    }
  }
`;

const BaseCurrency = props => {
  return (
    <StyledBaseCurrency>
      <p>USD - United States Dollars</p>
      <div className="header">
        <h1>USD</h1>
        <form className="ui form">
          <div className="ui massive transparent input">
            <input
              type="number"
              placeholder="Enter USD"
              value={props.usdInput}
              onChange={e => props.onUSDChange(e)}
            />
          </div>
        </form>
      </div>
    </StyledBaseCurrency>
  );
};

export default BaseCurrency;
