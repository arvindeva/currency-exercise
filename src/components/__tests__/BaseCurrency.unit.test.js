import React from 'react';
import { shallow, mount } from 'enzyme';
import BaseCurrency from '../BaseCurrency';

describe('BaseCurrency tests', () => {
  test('renders BaseCurrency', () => {
    const wrapper = shallow(<BaseCurrency />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Fires onUSDChange correctly', () => {
    const onUSDChange = jest.fn();
    const baseCurrency = mount(<BaseCurrency onUSDChange={onUSDChange} />);
    expect(baseCurrency.find('input').simulate('change'));
    expect(onUSDChange).toHaveBeenCalledTimes(1);
  });
});
