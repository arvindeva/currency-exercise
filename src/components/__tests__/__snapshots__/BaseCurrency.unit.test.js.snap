// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCurrency tests renders BaseCurrency 1`] = `
<styled.div>
  <p>
    USD - United States Dollars
  </p>
  <div
    className="header"
  >
    <h1>
      USD
    </h1>
    <form
      className="ui form"
    >
      <div
        className="ui massive transparent input"
      >
        <input
          onChange={[Function]}
          placeholder="Enter USD"
          type="number"
        />
      </div>
    </form>
  </div>
</styled.div>
`;
